### PR TITLE
Added thread-Id in logs

### PR DESCRIPTION
**Pull request by li-pkhandelwal**
Log snippet
**Earlier**
2020/04/17 16:31:12.906 +0530 INFO [JdbcTriggerImpl] [Azkaban] Updating trigger 5 into db.
2020/04/17 16:32:12.018 +0530 INFO [ExecutionFlowDao] [Azkaban] ExecutionFlow polling lock value: true for executorId: 98
2020/04/17 16:32:12.020 +0530 INFO [ExecutionFlowDao] [Azkaban] Released polling lock for executorId: 98
2020/04/17 16:32:12.021 +0530 INFO [FlowRunnerManager] [Azkaban] Submitting flow 1142
2020/04/17 16:32:12.023 +0530 INFO [FlowPreparer] [Azkaban] Project ProjectVersion{projectId=1, version=3, installedDir=projects/1.3, dirSizeInByte=null, lastAccessTime=null} already cached. Skipping download. ExecId: 1142
2020/04/17 16:32:12.028 +0530 INFO [FlowPreparer] [Azkaban] Setting up execution dir executions/1142 took 0 sec(s)
2020/04/17 16:32:12.029 +0530 INFO [FlowPreparer] [Azkaban] Flow preparation completed in 0 sec(s), out ot which 0 sec(s) was spent inside critical section. [execid: 1142, path: executions/1142]
2020/04/17 16:32:12.029 +0530 INFO [FlowRunnerManager] [Azkaban] Configuring Azkaban metrics tracking for flow runner object
2020/04/17 16:32:12.041 +0530 INFO [ExecutorDao] [Azkaban] Uploading flowId valid_dag_2
2020/04/17 16:32:12.046 +0530 INFO [ExecutorDao] [Azkaban] Uploading flowId valid_dag_2,valid_dag_3:valid_dag_2:valid_dag_3
2020/04/17 16:32:12.052 +0530 INFO [ExecutorDao] [Azkaban] Uploading flowId valid_dag_2,valid_dag_3:valid_dag_2:valid_dag_3,valid_dag_4:valid_dag_2:valid_dag_3:valid_dag_4
2020/04/17 16:32:12.058 +0530 INFO [JobCallbackManager] [Azkaban] No callback property for SUCCEEDED, exec id: 1142

**After**
2020/05/06 00:33:17.993 +0530  INFO [JdbcTriggerImpl] [TriggerRunnerManager-Trigger-Scanner-Thread] [Azkaban] Updating trigger 5 into db.
2020/05/06 00:34:16.856 +0530  INFO [FlowRunnerManager] [pool-10-thread-1] [Azkaban] Submitting flow 3486
2020/05/06 00:34:16.857 +0530  INFO [FlowPreparer] [pool-10-thread-1] [Azkaban] Project ProjectVersion{projectId=1, version=3, installedDir=projects/1.3, dirSizeInByte=null, lastAccessTime=null} already cached. Skipping download. ExecId: 3486
2020/05/06 00:34:16.861 +0530  INFO [FlowPreparer] [pool-10-thread-1] [Azkaban] Setting up execution dir executions/3486 took 0 sec(s)
2020/05/06 00:34:16.861 +0530  INFO [FlowPreparer] [pool-10-thread-1] [Azkaban] Flow preparation completed in 0 sec(s), out ot which 0 sec(s) was spent inside critical section. [execid: 3486, path: executions/3486]
2020/05/06 00:34:16.861 +0530  INFO [FlowRunnerManager] [pool-10-thread-1] [Azkaban] Configuring Azkaban metrics tracking for flow runner object
2020/05/06 00:34:16.911 +0530  INFO [ExecutorDao] [JobRunner-task_1-3486] [Azkaban] Uploading flowId valid_dag_2
2020/05/06 00:34:16.917 +0530  INFO [ExecutorDao] [JobRunner-task_3-3486] [Azkaban] Uploading flowId valid_dag_2,valid_dag_3:valid_dag_2:valid_dag_3
2020/05/06 00:34:16.922 +0530  INFO [ExecutorDao] [JobRunner-task_4-3486] [Azkaban] Uploading flowId valid_dag_2,valid_dag_3:valid_dag_2:valid_dag_3,valid_dag_4:valid_dag_2:valid_dag_3:valid_dag_4
2020/05/06 00:34:16.929 +0530  INFO [ExecutorDao] [JobRunner-task_2-3486] [Azkaban] Uploading flowId valid_dag_2,emb_2:valid_dag_2:emb_2
2020/05/06 00:34:16.943 +0530  INFO [JobCallbackManager] [JobRunner-task_1-3486] [Azkaban] No callback property for SUCCEEDED, exec id: 3486

### CODE REVIEW

File: azkaban-common/src/test/resources/log4j.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 log4j.rootLogger=INFO, Console
 log4j.appender.Console=org.apache.log4j.ConsoleAppender
 log4j.appender.Console.layout=org.apache.log4j.PatternLayout
-log4j.appender.Console.layout.ConversionPattern=%d{yyyy/MM/dd HH:mm:ss.SSS Z} %p [%c{1}] %m%n
+log4j.appender.Console.layout.ConversionPattern=%d{yyyy/MM/dd HH:mm:ss.SSS Z} %5p [%F:%L] [%t] %m%n
```
Comment (li-ygerchikov) [inline azkaban-common/src/test/resources/log4j.properties:4]: Just curious, what's the reason for changing `%p` to `%5p`? I *think* it's going to add padding to the severity levels shorter than 5 characters (INFO, WARN). Why do we want it?

**Comment by ypadron-in** [comment]
These changes should not affect the inspection of logs that currently takes place in Azkaban code to kill jobs in Yarn or construct the url to job logs in the Hadoop/Spark servers, but I agree with other folks in making sure there is no significant performance impact involved.

File: azkaban-common/src/test/resources/log4j.properties (hunk 1 above)
Comment (li-pkhandelwal) [reply azkaban-common/src/test/resources/log4j.properties:4]: Yes, this adds padding such that information in log line start from same column(as ERROR, DEBUG are 5 chars) in line number

**Comment by li-pkhandelwal** [comment]
> Thanks for looking into this. The additional information will be super helpful if we can put this in place.
> 
> One thing that I would like to get an idea about is the performance impact of getting this additional information. Please think how can we measure the impact? Should we roll-in the changes into a few executors of a few clusters?
> 
> What I am worried about is that the slowdown of logging will result in bloating of IO buffers. This increases the native memory of the JVM, which we already see as a problem. We would like to make sure that this doesn't slowdown routing and increase memory usage to an unacceptable point.
> 
> NOTE:
> 'F', 'L' and 'M' have can have a negative impact on performance. This change is choosing two of those 'F' and 'L'
> 
> Reference: https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html

For printing line number, JVM computes the line number from .class file. In my understanding, this would be a CPU intensive process and IO buffer should not be affected as the log would make it to IO buffer only when computation of line number completes which does not alter the current handling for printing logs.
However, will update the changes to show only thread_id in logs as that is the more important information in logs and defer the change for printing line number after measuring impact on performance.

**Comment by li-pkhandelwal** [comment]
> Title will have to be updated to reflect the changed scope.



> Title will have to be updated to reflect the changed scope.

Done

**Pull request merged**